spec/plugins/mosaic/schemas/MosaicDefinitionTransactionBodyDTO.yml

```diff
@@ -12,9 +12,8 @@ properties:
     $ref: "../../../schemas/BlockDuration.yml"
   nonce:
     type: integer
-    format: int64
+    format: uint32
```

Comment:
There is no format uint32 in the OpenAPI spec: https://swagger.io/specification/#data-types

Note that uint32 are in range between 0 to 4,294,967,295, where as int32 range varies between -2,147,483,648 to 2,147,483,647. To be able to represent all uint32 values, it's common to define the integer as int64 (-9223372036854775808 to 9223372036854775807).

Reply:
ah i think i remember this discussion now ^^

Reply:
@dgarcia360 can't we have int32? int/uint treatment methods between those is non-existant compared to the differences that our system imposes when treating (any kind of) int and int64, which has a dynamic custom representation

almost every (if not all) of the integers in the project/API right now are supposedly unsigned and they are documented in the openapi as regular integers

take a look at MosaicFlags that are int32 in openapi https://github.com/nemtech/symbol-openapi/blob/master/spec/plugins/mosaic/schemas/MosaicFlagsEnum.yml#L2 where in catbuffer they are uint8 https://github.com/nemtech/catbuffer/blob/master/schemas/mosaic/mosaic_types.cats#L4

it is true that a uint32 does not fit into an int32 from a semantic point of view, but technically, the bits of a uint32 fit into an int32, the only thing that changes is how data is interpreted - and we are giving importance to the bytes, not the definitive value (specially in the nonce case, no one cares if it is 23 milions, it's just about the bits)

it's not strict to treat it as int32 because semantically it wouldn't fit (though it makes no sense), but it's not strict either to treat it as int64 because it isn't, and right now it is screwing us

for all that matters, we are using openapi to determine what kind of parsers are being used, and right now in the project overall we have two kinds of number parsing methods: 1) Int64 2) all of the numbers minus Int64 - i'd say it makes more sense to treat unsigneds as "all the other numbers" rather than putting them into the Int64 category, because they do not require the special parsing Int64 needs

Reply:
i would agree with @Vektrat on changing to int32. 

background is we are facing an issue of `nonce` fields being displayed as signed numbers which breaks the SDK's expected formatting.

SDK awaits a *hex* (because "int64") but REST returns a signed integer (because "int64") **but, and this is the problem**, in database REST holds uint32. So i'd agree going for _int32_ makes sense given the non-utility of the actual number it holds.

Reply:
I removed here the format, now should be all integers should be int32 by default. Let me know if this works for you https://github.com/nemtech/symbol-openapi/pull/90

Reply:
as long as you agree that this change is not going backwards in the openapi project works out for me, you know how this works better anyway, i'll leave PR acceptance to you/others